ProcessLib/TwoPhaseFlowWithPrho/CreateTwoPhaseFlowWithPrhoProcess.cpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+#include "CreateTwoPhaseFlowWithPrhoProcess.h"
+#include <cassert>
+#include "BaseLib/Functional.h"
+#include "MeshLib/MeshGenerators/MeshGenerator.h"
+#include "ProcessLib/Parameter/ConstantParameter.h"
+#include "ProcessLib/TwoPhaseFlowWithPrho/CreateTwoPhaseFlowPrhoMaterialProperties.h"
+#include "ProcessLib/TwoPhaseFlowWithPrho/TwoPhaseFlowWithPrhoMaterialProperties.h"
+#include "ProcessLib/Utils/ParseSecondaryVariables.h"
+#include "ProcessLib/Utils/ProcessUtils.h"
+#include "TwoPhaseFlowWithPrhoProcess.h"
+#include "TwoPhaseFlowWithPrhoProcessData.h"
+
+namespace ProcessLib
+{
+namespace TwoPhaseFlowWithPrho
+{
+std::unique_ptr<Process> CreateTwoPhaseFlowWithPrhoProcess(
+    MeshLib::Mesh& mesh,
+    std::unique_ptr<ProcessLib::AbstractJacobianAssembler>&& jacobian_assembler,
+    std::vector<ProcessVariable> const& variables,
+    std::vector<std::unique_ptr<ParameterBase>> const& parameters,
+    unsigned const integration_order,
+    BaseLib::ConfigTree const& config,
+    std::map<std::string,
+             std::unique_ptr<MathLib::PiecewiseLinearInterpolation>> const&
+        curves)
+{
+    //! \ogs_file_param{prj__processes__process__type}
+    config.checkConfigParameter("type", "TWOPHASE_FLOW_PRHO");
+
+    DBUG("Create TwoPhaseFlowProcess with Prho model.");
+    //! \ogs_file_param{prj__processes__process__TWOPHASE_FLOW_PRHO__process_variables}
+    auto const pv_config = config.getConfigSubtree("process_variables");
+
+    auto process_variables = findProcessVariables(
+        variables, pv_config,
+        {//! \ogs_file_param_special{prj__processes__process__TWOPHASE_FLOW_PRHO__process_variables__gas_pressure}
+         "liquid_pressure",
+         //! \ogs_file_param_special{prj__processes__process__TWOPHASE_FLOW_PRHO__process_variables__overall_mass_density}
+         "overall_mass_density"});
+
+    SecondaryVariableCollection secondary_variables;
+
+    NumLib::NamedFunctionCaller named_function_caller(
+        {"TwoPhaseFlow_pressure"});
+
+    ProcessLib::parseSecondaryVariables(config, secondary_variables,
+                                        named_function_caller);
+    // Specific body force
+    Eigen::VectorXd specific_body_force;
+
+    std::vector<double> const b =
+        //! \ogs_file_param{prj__processes__process__TWOPHASE_FLOW_PRHO__specific_body_force}
+        config.getConfigParameter<std::vector<double>>("specific_body_force");
+    assert(b.size() > 0 && b.size() < 4);
+    specific_body_force.resize(b.size());
```

Comment:
You can create the `specific_body_force` vector with correct size. See documentation of the available constructors http://eigen.tuxfamily.org/dox/group__QuickRefPage.html